samples/api/hpp_hello_triangle/hpp_hello_triangle.cpp

```diff
@@ -0,0 +1,1004 @@
+{
+	if (ext == "vert")
+	{
+		return VK_SHADER_STAGE_VERTEX_BIT;
+	}
+	else if (ext == "frag")
+	{
+		return VK_SHADER_STAGE_FRAGMENT_BIT;
+	}
+	else if (ext == "comp")
+	{
+		return VK_SHADER_STAGE_COMPUTE_BIT;
+	}
+	else if (ext == "geom")
+	{
+		return VK_SHADER_STAGE_GEOMETRY_BIT;
+	}
+	else if (ext == "tesc")
+	{
+		return VK_SHADER_STAGE_TESSELLATION_CONTROL_BIT;
+	}
+	else if (ext == "tese")
+	{
+		return VK_SHADER_STAGE_TESSELLATION_EVALUATION_BIT;
+	}
+
+	throw std::runtime_error("No Vulkan shader stage found for the file extension name.");
+};
+
+/**
+ * @brief Initializes the Vulkan instance.
+ *
+ * @param context A newly created Vulkan context.
+ * @param required_instance_extensions The required Vulkan instance extensions.
+ * @param required_validation_layers The required Vulkan validation layers
+ */
+void HPPHelloTriangle::init_instance(Context &                        context,
+                                     const std::vector<const char *> &required_instance_extensions,
+                                     const std::vector<const char *> &required_validation_layers)
+{
+	LOGI("Initializing vulkan instance.");
+
+	static vk::DynamicLoader  dl;
+	PFN_vkGetInstanceProcAddr vkGetInstanceProcAddr =
+	    dl.getProcAddress<PFN_vkGetInstanceProcAddr>("vkGetInstanceProcAddr");
+	VULKAN_HPP_DEFAULT_DISPATCHER.init(vkGetInstanceProcAddr);
+
+	std::vector<vk::ExtensionProperties> instance_extensions = vk::enumerateInstanceExtensionProperties();
+
+	std::vector<const char *> active_instance_extensions(required_instance_extensions);
+
+#if defined(VKB_DEBUG) || defined(VKB_VALIDATION_LAYERS)
+	active_instance_extensions.push_back(VK_EXT_DEBUG_REPORT_EXTENSION_NAME);
+#endif
+
+#if defined(VK_USE_PLATFORM_ANDROID_KHR)
+	active_instance_extensions.push_back(VK_KHR_ANDROID_SURFACE_EXTENSION_NAME);
+#elif defined(VK_USE_PLATFORM_WIN32_KHR)
+	active_instance_extensions.push_back(VK_KHR_WIN32_SURFACE_EXTENSION_NAME);
+#elif defined(VK_USE_PLATFORM_MACOS_MVK)
+	active_instance_extensions.push_back(VK_MVK_MACOS_SURFACE_EXTENSION_NAME);
+#elif defined(VK_USE_PLATFORM_XCB_KHR)
+	active_instance_extensions.push_back(VK_KHR_XCB_SURFACE_EXTENSION_NAME);
+#elif defined(VK_USE_PLATFORM_DISPLAY_KHR)
+	active_instance_extensions.push_back(VK_KHR_DISPLAY_EXTENSION_NAME);
+#else
+#	pragma error Platform not supported
+#endif
+
+	if (!validate_extensions(active_instance_extensions, instance_extensions))
+	{
+		throw std::runtime_error("Required instance extensions are missing.");
+	}
+
+	std::vector<vk::LayerProperties> supported_validation_layers = vk::enumerateInstanceLayerProperties();
+
+	std::vector<const char *> requested_validation_layers(required_validation_layers);
+
+#ifdef VKB_VALIDATION_LAYERS
+	// Determine the optimal validation layers to enable that are necessary for useful debugging
+	std::vector<const char *> optimal_validation_layers = vkb::get_optimal_validation_layers(supported_validation_layers);
```

Comment:
This doesn't compile for me when enabling validation layers. The `get_optimal_validation_layers` function takes a vector if `VkLayerProperties`, which is not compatible with the hpp's `vk::LayerProperties` array. 

I guess we need an additional overload for this to work.

Reply:
Resolved with https://github.com/KhronosGroup/Vulkan-Samples/pull/294/commits/3e62a9f5a339e539e4833c0594e0a8376991286d.